third_party/2and3/boto/utils.pyi

```diff
@@ -0,0 +1,215 @@
+    command = ...  # type: subprocess._CMD
+    log_fp = ...  # type: IO
+    wait = ...  # type: bool
+    fail_fast = ...  # type: bool
+
+    def __init__(
+        self,
+        command: subprocess._CMD,
+        wait: bool = ...,
+        fail_fast: bool = ...,
+        cwd: Optional[subprocess._TXT] = ...,
+    ) -> None: ...
+
+    process = ...  # type: subprocess.Popen
+
+    def run(self, cwd: Optional[subprocess._CMD] = ...) -> Optional[int]: ...
+    def setReadOnly(self, value) -> None: ...
+    def getStatus(self) -> Optional[int]: ...
+
+    status = ...  # type: Optional[int]
+
+    def getOutput(self) -> str: ...
+
+    output = ...  # type: str
+
+class AuthSMTPHandler(logging.handlers.SMTPHandler):
+    username = ...  # type: str
+    password = ...  # type: str
+    def __init__(
+        self,
+        mailhost: str,
+        username: str,
+        password: str,
+        fromaddr: str,
+        toaddrs: Sequence[str],
+        subject: str,
+    ) -> None: ...
+    def emit(self, record: logging.LogRecord) -> None: ...
+
+class LRUCache(Dict[_KT, _VT]):
+    class _Item:
+        previous = ...  # type: Optional[LRUCache._Item]
+        key = ...
+        value = ...
+        def __init__(self, key, value) -> None: ...
+
+    capacity = ...  # type: int
+    head = ...  # type: Optional[LRUCache._Item]
+    tail = ...  # type: Optional[LRUCache._Item]
+
+    def __init__(self, capacity: int) -> None: ...
+
+
+# This exists to work around Password.str's name shadowing the str type
+_str = str
+
+class Password:
+    hashfunc = ...  # type: Callable
+    str = ...  # type: Optional[_str]
+
+    def __init__(
+        self,
+        str: Optional[_str] = ...,
+        hashfunc: Optional[Callable] = ...,
+    ) -> None: ...
+    def set(self, value: Union[bytes, _str]) -> None: ...
+    def __str__(self) -> _str: ...
+    def __eq__(self, other: Any) -> bool: ...
+    def __len__(self) -> int: ...
+
+def notify(
+    subject: str,
+    body: Optional[str] = ...,
+    html_body: Optional[Union[Sequence[str], str]] = ...,
+    to_string: Optional[str] = ...,
+    attachments: Optional[Iterable] = ...,
+    append_instance_id: bool = ...,
+) -> None: ...
+def get_utf8_value(value: str) -> str: ...
+def mklist(value: Any) -> List: ...
+def pythonize_name(name: str) -> str: ...
+def write_mime_multipart(
+    content: List[Tuple[str, str]],
+    compress: bool = ...,
+    deftype: str = ...,
+    delimiter: str = ...,
+) -> str: ...
+def guess_mime_type(content: str, deftype: str) -> str: ...
+def compute_md5(
+    fp: IO,
+    buf_size: int = ...,
+    size: Optional[int] = ...,
+) -> Tuple[str, str, int]: ...
+def compute_hash(
+    fp: IO,
+    buf_size: int = ...,
+    size: Optional[int] = ...,
+    hash_algorithm: Any = ...,
+) -> Tuple[str, str, int]: ...
+def find_matching_headers(name: str, headers: Dict[str, str]) -> List[str]: ...
+def merge_headers_by_name(name: str, headers: Dict[str, str]) -> str: ...
```

Comment:
Mapping, and the values can be `Optional[str]`.

Reply:
Perhaps I should change the value of all relevant `headers` variable to be `Optional[str]`?

Reply:
I traced through the code and believe it is safe to do it for `merge_headers_by_name`, `find_matching_headers`, and `canonical_string` but not the others so I will go forward with that change

Reply:
Sounds good